.github/actions/build-sample-app/action.yml

```diff
@@ -0,0 +1,142 @@
+name: 'Build Sample App'
+description: 'Builds and publishes a sample app'
+inputs:
+  app-dir:
+    description: 'relative path to the app directory'
+    required: true
+  mvn-build-commands:
+    description: 'commands passed to \"mvnw" to build the app'
+    required: false
+    default: '-B clean verify'
+  artifactory-publish:
+    description: 'whether or not to publish to artifactory'
+    type: boolean
+    required: false
+    default: true
+  jf-artifactory-spring:
+    description: 'encoded JFrog server id configuration'
+    required: false
+  jf-mvn-build-commands:
+    description: 'commands passed to \"jfrog rt mvn\" to publish the app'
+    required: false
+    default: '-B clean install -DskipTests'
+  docker-push:
+    description: 'whether or not to push docker image to docker hub'
+    type: boolean
+    required: false
+    default: true
+  docker-registry:
+    description: 'docker registry'
+    required: false
+  docker-username:
+    description: 'docker username'
+    required: false
+  docker-password:
+    description: 'docker password'
+    required: false
+  docker-images:
+    description: 'csv of docker images to push'
+    required: false
+  docker-images-override:
+    description: 'csv of docker image tags to use when pushing (parallel array to docker-images)'
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - uses: jvalkeal/setup-maven@v1
+      with:
+        maven-version: 3.8.4
+    - uses: actions/setup-java@v2
+      with:
+        distribution: adopt
+        java-version: 8
+        cache: maven
+
+    - name: Maven build
+      shell: bash
+      run: ./mvnw ${{ inputs.mvn-build-commands }}
+      working-directory: ${{ inputs.app-dir }}
+
+    - name: Setup JFrog CLI
+      if: ${{ inputs.artifactory-publish == 'true' }}
+      uses: jfrog/setup-jfrog-cli@v1
+      with:
+        version: 1.46.4
+      env:
+        JF_ARTIFACTORY_SPRING: ${{ inputs.jf-artifactory-spring }}
+
+    - name: Configure JFrog Cli
+      if: ${{ inputs.artifactory-publish == 'true' }}
```

Comment:
Breaking this action down into 3 smaller actions (build, publish, push) would do the following: 
1. remove need to have the if guard on multiple steps in the file (there would just be one guard on the sub-action callout)
2. keep the actions focused (this one is getting complicated and busy)
3. allow us to force required params when the sub-action is enabled. Currently we have to make all the things non-required in case they don't enable the sub-action. 
The "possible improvement" listed in the PR description would remove the need to have these if guards across all of the related steps in this file.

Reply:
This is now included in this propsoal.